fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaProgress.java

```diff
@@ -126,16 +126,22 @@ private void getReadableProgress(Map<Integer, String> showPartitionIdToOffset) {
     // modify the partition offset of this progress.
     // throw exception is the specified partition does not exist in progress.
     public void modifyOffset(List<Pair<Integer, Long>> kafkaPartitionOffsets) throws DdlException {
-        // kafka progress will not initialized & update if current role is follower
-        boolean notInitialized = partitionIdToOffset.isEmpty();
         for (Pair<Integer, Long> pair : kafkaPartitionOffsets) {
-            if (!notInitialized && !partitionIdToOffset.containsKey(pair.first)) {
+            if (!partitionIdToOffset.containsKey(pair.first)) {
                 throw new DdlException("The specified partition " + pair.first + " is not in the consumed partitions");
             }
         }
         for (Pair<Integer, Long> pair : kafkaPartitionOffsets) {
             partitionIdToOffset.put(pair.first, pair.second);
         }
+        // update kafkaPartitionOffsets as well, so that the current partitonIdToOffset can be completely persisted
```

Comment:
1. Why change the kafkaPartitionOffsets in the parameter? I think the parameter should keep unchanged.
2. In commit message, add the solution and results besides the problem description.

Reply:
1. To make sure that the parameter is the same with the actual value so that the actual value can be persist.
2. I think I did. I have describe the behavior in my commit on all issues.
But I get your point, I'll add more comments here.

Reply:
If kafkaPartitionOffsets is the same as partitionIdToOffset, why not copy it directly instead of a loop?

Reply:
It's not. There's an example described in the comments.
```
    //   if current offset is {0:11, 1:22, 2:33, 3:55}, the parameter `kafkaPartitionOffsets` is {0: 22, 2:44}
    //   in this function, current offset is modified as {0: 22, 1: 22, 2:44, 3:55}
    //   so is the parameter `kafkaPartitionOffsets`.
```